include/bluetooth/scan.rst

```diff
@@ -98,6 +98,31 @@ Check the following table for the details on the behavior of the available filte
 |              | If not all of these types match, the ``not found`` callback is triggered.                                 |
 +--------------+-----------------------------------------------------------------------------------------------------------+
 
+Connection attempts filter
+==========================
+
+After a scanning is started by the application some peripheral devices can disconnect immediately for some reasons during ongoing scanning session.
+This may result in a loop of the connection and disconnection events.
+To avoid this loop you can enable the Connection attempts filter which limits number of the connection tries.
+Use the :option:`CONFIG_BT_SCAN_CONNECTION_ATTEMPTS_FILTER` to enable this filter.
+This filter automatically tracks connected devices and count all disconnection events for given devices if disconnection counts is greater than number of allowed attempts the scanning module ignores this device.
+Filtered devices must must be removed manually from filter array using the :cpp:func:`bt_scan_attempts_filter_clean`.
+It is recommended to use the :cpp:func:`bt_scan_attempts_filter_clean` before before each scan starts unless the yours application has different requirements.
+If the filter array is full, the scanning module overwrites the oldest device with the new one.
```

Comment:
Overwriting the oldest device might not be the best strategy.
The least-recently-used (or seen) device would be better in my opinion.

An aggressive advertiser might be taken in and put back into the list continuously, while an advertiser that has moved out of range will be in the list forever.

Reply:
I think that overwriting the oldest device is simpler (we do not need to track the least-recently-used device) and its good enough (at least for nrf_desktop usecase). The only drawback could be that we could remove nearby advertiser from the list for a moment, but it would be added back after a few unsuccessful connection tries - so it's not causing any problems for us. 

With both approaches we need to assume that our list of devices is big enough - otherwise we won't be able to keep all the nearby advertising devices and we would still try to connect to one of them.

Reply:
Ok, thank you @MarekPieta. Let's keep it then.